src/core/model.jl

```diff
@@ -62,11 +62,20 @@ function connect_parameter(m::Model, dst::Pair{Symbol, Symbol}, src::Pair{Symbol
     connect_parameter(m.md, dst[1], dst[2], src[1], src[2], backup; ignoreunits=ignoreunits, offset=offset)
 end
 
-function set_external_param!(m::Model, name::Symbol, value::ModelParameter)
+function set_external_param!(m::Model, name::Symbol, value::ModelParameter; param_dims = [], num_dims = 0)
     set_external_param!(m.md, name, value)
     decache(m)
 end
 
+function set_external_param!(m::Model, name::Symbol, value::Number; param_dims = [], num_dims = 0)
+    set_external_param!(m.md, name, value)
+    decache(m)
+end
+
+function set_external_param!(m::Model, name::Symbol, value::Union{AbstractArray, Range, Tuple}; param_dims = [], num_dims = 0)
```

Comment:
Compute `param_dims` and `num_dims` here rather than passing as args. We need to consider whether setting an external parameter should invalidate the model instance (i.e., if we should call `decache(m)` here. I'll have to review the MCS code but I think we can avoid invalidation when updating a value, but not if we're adding a new external parameter.

Reply:
I was thinking I would calculate it within ...

`set_external_param!(md::ModelDef, name::Symbol, value::Union{AbstractArray, Range, Tuple})`

... since that's the only place we actually end up needing those parameters